src/pybind/mgr/dashboard/frontend/src/app/ceph/block/mirroring/pool-edit-peer-modal/pool-edit-peer-modal.component.ts

```diff
@@ -0,0 +1,134 @@
+import { Component, OnInit } from '@angular/core';
+import { AbstractControl, FormControl, Validators } from '@angular/forms';
+
+import { BsModalRef } from 'ngx-bootstrap/modal';
+
+import { RbdMirroringService } from '../../../../shared/api/rbd-mirroring.service';
+import { CdFormGroup } from '../../../../shared/forms/cd-form-group';
+import { FinishedTask } from '../../../../shared/models/finished-task';
+import { TaskWrapperService } from '../../../../shared/services/task-wrapper.service';
+import { PoolEditPeerResponseModel } from './pool-edit-peer-response.model';
+
+@Component({
+  selector: 'cd-pool-edit-peer-modal',
+  templateUrl: './pool-edit-peer-modal.component.html',
+  styleUrls: ['./pool-edit-peer-modal.component.scss']
+})
+export class PoolEditPeerModalComponent implements OnInit {
+  mode: string;
+  poolName: string;
+  peerUUID: string;
+
+  editPeerForm: CdFormGroup;
+  bsConfig = {
+    containerClass: 'theme-default'
+  };
+  pattern: string;
+
+  response: PoolEditPeerResponseModel;
+
+  constructor(
+    public modalRef: BsModalRef,
+    private rbdMirroringService: RbdMirroringService,
+    private taskWrapper: TaskWrapperService
+  ) {
+    this.createForm();
+  }
+
+  createForm() {
+    this.editPeerForm = new CdFormGroup({
+      clusterName: new FormControl('', {
+        validators: [Validators.required, this.validateClusterName]
+      }),
+      clientID: new FormControl('', {
+        validators: [Validators.required, this.validateClientID]
+      }),
+      monAddr: new FormControl('', {
+        validators: [this.validateMonAddr]
+      }),
+      key: new FormControl('', {
+        validators: [this.validateKey]
+      })
+    });
+  }
+
+  ngOnInit() {
+    this.pattern = `${this.poolName}/${this.peerUUID}`;
+    if (this.mode === 'edit') {
+      this.rbdMirroringService
+        .getPeer(this.poolName, this.peerUUID)
+        .subscribe((resp: PoolEditPeerResponseModel) => {
+          this.setResponse(resp);
+        });
+    }
+  }
+
+  validateClusterName(control: AbstractControl) {
+    if (!control.value.match(/^[\w\-_]*$/)) {
+      return { invalidClusterName: { value: control.value } };
+    }
+  }
+
+  validateClientID(control: AbstractControl) {
+    if (!control.value.match(/^(?!client\.)[\w\-_.]*$/)) {
+      return { invalidClientID: { value: control.value } };
+    }
+  }
+
+  validateMonAddr(control: AbstractControl) {
+    if (!control.value.match(/^[,; ]*([\w\-_\[\]]+(:[\d]+)?[,; ]*)*$/)) {
```

Comment:
Shouldn't we also accept IP addresses?

![screenshot from 2018-12-14 11-17-48](https://user-images.githubusercontent.com/14297426/50000653-1d7f7300-ff92-11e8-8eff-9deacc9fd338.png)

Reply:
Thx -- fixed.